aiohttp/pytest_plugin.py

```diff
@@ -119,29 +119,19 @@ def pytest_configure(config):
     LOOP_FACTORIES.clear()
     LOOP_FACTORY_IDS.clear()
 
-    if loops:
-        for names in (name.split(',') for name in loops):
-            for name in names:
-                name = name.strip()
-                if name not in factories:
-                    raise ValueError(
-                        "Unknown loop '%s', available loops: %s" % (
-                            name, list(factories.keys())))
-
-                LOOP_FACTORIES.append(factories[name])
-                LOOP_FACTORY_IDS.append(name)
-    else:
-        LOOP_FACTORIES.append(asyncio.new_event_loop)
-        LOOP_FACTORY_IDS.append('pyloop')
-
-        if uvloop is not None:  # pragma: no cover
-            LOOP_FACTORIES.append(uvloop.new_event_loop)
-            LOOP_FACTORY_IDS.append('uvloop')
-
-        if tokio is not None:
-            LOOP_FACTORIES.append(tokio.new_event_loop)
-            LOOP_FACTORY_IDS.append('tokio')
-
+    if loops == 'all':
+        loops = 'pyloop,uvloop?,tokio?'
+
+    for name in loops.split(','):
+        required = not name.endswith('?')
+        name = name.strip(' ?')
+        if name in factories:
+            LOOP_FACTORIES.append(factories[name])
+            LOOP_FACTORY_IDS.append(name)
+        elif required:
+            raise ValueError(
```

Comment:
I tried adding a test to add coverage for this case:

```python
def test_wrong_loop(testdir, capsys):
    testdir.makepyfile("""\
import asyncio
pytest_plugins = 'aiohttp.pytest_plugin'

@asyncio.coroutine
def test_foobar():
    assert 1 == 1
""")
    result = testdir.runpytest('-p', 'no:sugar', '--loop', 'foobar')
    print(result.outlines)
```

But I get back an error saying the `--loop` argument is not recognised, I get the same error if I use a real loop eg. `pyloop` so I think this is a limitation of the pytest self test fixtures.

Reply:
it wasn't covered before.

Reply:
Ok, let's leave these lines untested. I could live with it.